### PR TITLE
Resolves #2962 - Add properties config to Maven parameters

### LINKED ISSUE #2962: [REQ] Expose all top-level config options as properties
### Is your feature request related to a problem? Please describe.

Some of the options in the CodeGenMojo are exposed such that you can pass the property on the Maven command-line using `-Dopenapi.generator.maven.plugin.<property>`, but not all. I propose that ALL of those properties should be exposed and that the documentation should be updated to reflect the property name as well.

## Describe the solution you'd like

Modify the CodeGenMojo to include property exposure in all @Parameter annotations

## Describe alternatives you've considered

None that I can think of

## Additional context

I can submit a PR for this if it is helpful.

### PR DESCRIPTION
**Pull request by InfoSec812**
### Description of the PR

Exposes the CodeGenMojo configuration settings as java properties to that they can be modified using `-D` arguments

Resolves #2962

### CODE REVIEW

File: modules/openapi-generator-maven-plugin/README.md [hunk 1]
```diff
@@ -101,7 +106,7 @@ Specifying a custom generator is a bit different. It doesn't support the classpa
             </goals>
             <configuration>
                 <inputSpec>${project.basedir}/src/main/resources/yaml/yamlfilename.yaml</inputSpec>
-                <!-- language file, like e.g. JavaJaxRSCodegen -->
+                <!- language file, like e.g. JavaJaxRSCodegen -->
```
Comment (wing328) [inline modules/openapi-generator-maven-plugin/README.md]: @InfoSec812 the dash seems to be accidentally removed?

Comment (InfoSec812) [reply modules/openapi-generator-maven-plugin/README.md]: Whoops! Good catch. Must have accidentally done that while doing a find/replace to update the docs. Ill fix it now.

**Comment by wing328** [comment]
@InfoSec812 thanks for the PR. When you've time, I wonder if you can resolve the merge conflicts and then I'll merge this PR into master.

**Comment by InfoSec812** [comment]
> @InfoSec812 thanks for the PR. When you've time, I wonder if you can resolve the merge conflicts and then I'll merge this PR into master.

Resolved and I included an exposed property for the `engine` parameter

**Comment by rvowles** [comment]
There appears to be a bug in this PR. I just build master and am using it, and its exploding with this field as it does not match its named parameter. Maven is trying to give it a default value by setting the field "strictSpec" and the field is called "strictSpecBehaviour".  @InfoSec812  @wing328

**Comment by wing328** [comment]
@rvowles please open an issue to track it and we'll look into it.

**Pull request merged**